frappe/public/js/frappe/views/kanban/kanban_board.js

```diff
@@ -640,7 +652,7 @@ frappe.provide("frappe.views");
 			assigned_list: card.assigned_list || assigned_list,
 			comment_count: card.comment_count || comment_count,
 			color: card.color || null,
-			doc: doc
+			doc: doc || card
```

Comment:
why?

Reply:
The `doc` object was coming up `undefined` for every task on the board, and I couldn't figure out the cause for that (also, possibly related, but [choosing a color for a task does not display it in the Kanban view](https://github.com/frappe/frappe/issues/6522)).

However, the document data was present in the `card` object, so I picked that up.